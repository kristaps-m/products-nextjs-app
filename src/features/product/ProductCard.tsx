import { Product } from "../../app/models/product";

interface Props {
  product: Product;
}

export default function ProductCard({ product }: Props) {
  return (
    <div
      data-testid="productCard-1"
      className="p-3 m-3 bg-white rounded-lg shadow-lg overflow-hidden"
    >
      <img
        src={`https://picsum.photos/id/${product.id + 10}/200/300`}
        alt={product.name}
        className="w-full h-48 object-cover"
      />
      <div className="p-4">
        <h3 className="text-xl font-bold mb-2">{product.name}</h3>
        <p className="text-gray-600 text-sm">
          Price: ${product.price.toFixed(2)}
        </p>
        <p className="text-gray-600 text-sm">Category: {product.category}</p>
        <a href={`/products/${product.id}`}>
          <button className="mt-2 px-4 py-2 bg-blue-500 hover:bg-blue-600 text-white rounded-full">
            View
          </button>
        </a>
      </div>
    </div>
  );
}
