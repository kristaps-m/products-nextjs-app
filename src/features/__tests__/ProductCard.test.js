import '@testing-library/jest-dom'
import {render, screen, cleanup} from '@testing-library/react';
import ProductCard from '../product/ProductCard'
import TestRenderer from 'react-test-renderer';

afterEach(() => {
  cleanup();
});

test('should render ProductCard feature/product', () => {
  const product = { "id": 1,
    "name": "Apple",
    "price": 1.10,
    "currency": "EUR",
    "category": "Fruits",
    "description": "Fresh and delicious apples from the local orchard."}

  render(<ProductCard product={product}/>);
  const productCardElement = screen.getByTestId('productCard-1');

  expect(productCardElement).toBeInTheDocument();
  expect(productCardElement).toHaveTextContent('Price: $');
  expect(productCardElement).toHaveTextContent('Category:');
  expect(productCardElement).toHaveTextContent("View");
})

test('maches snapshot', () => {
  const product = { "id": 1,
    "name": "Apple",
    "price": 1.10,
    "currency": "EUR",
    "category": "Fruits",
    "description": "Fresh and delicious apples from the local orchard."}
  const tree = TestRenderer.create(<ProductCard product={product}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
