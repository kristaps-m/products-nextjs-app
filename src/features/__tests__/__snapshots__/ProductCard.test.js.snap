// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maches snapshot 1`] = `
<div
  className="p-3 m-3 bg-white rounded-lg shadow-lg overflow-hidden"
  data-testid="productCard-1"
>
  <img
    alt="Apple"
    className="w-full h-48 object-cover"
    src="https://picsum.photos/id/11/200/300"
  />
  <div
    className="p-4"
  >
    <h3
      className="text-xl font-bold mb-2"
    >
      Apple
    </h3>
    <p
      className="text-gray-600 text-sm"
    >
      Price: $
      1.10
    </p>
    <p
      className="text-gray-600 text-sm"
    >
      Category: 
      Fruits
    </p>
    <a
      href="/products/1"
    >
      <button
        className="mt-2 px-4 py-2 bg-blue-500 hover:bg-blue-600 text-white rounded-full"
      >
        View
      </button>
    </a>
  </div>
</div>
`;
